datm/cime_config/buildnml

```diff
@@ -205,8 +232,12 @@ def _create_namelists(case, confdir, inst_string, infile, nmlgen, data_list_path
     bias_correct = nmlgen.get_value("bias_correct")
     if bias_correct is not None:
         streamlist.append(bias_correct)
+
+    # Set anomaly forcing in datm.streams.xml
     anomaly_forcing = nmlgen.get_value("anomaly_forcing")
-    if anomaly_forcing[0] is not None:
+    if anomaly_forcing[0] is not None and anomaly_forcing[0] != "none":
+        if not mismatch_ok:
+            _check_datm_af_mismatch(anomaly_forcing[0], datm_mode)
```

Comment:
@ekluzek I've now added a check that DATM_MODE matches the reanalysis used to make the anomaly forcings. What do you think?

Reply:
I think this is great as a start. Longer term I'd like to see the AF forcing type to be part of the XML data in some form. But, since there's only one option having it hardcoded in the python is fine for now.

I like how you've done it with a subroutine which would allow unit-testing later. And I think the code here is well done and clear on what's happening.

Reply:
- [ ] Remove `mismatch_ok` option and instead provide instructions on how to fix error

Reply:
Done with 8aab1ae.